src/components/content/credential/Credential.tsx

```diff
@@ -43,7 +43,50 @@ function Credential(): JSX.Element {
     });
     const oidcIdToken: OidcIdToken = useOidcIdToken();
 
-    const columns: ColumnsType<AbstractCredentialInfo> = [
+    const userName = getUserName(oidcIdToken.idTokenPayload as object);
+    const credentialsQuery = useQuery({
+        queryKey: ['credentialsQuery', userName],
+        queryFn: () => CredentialsManagementService.getCredentialsByUser(userName),
+        staleTime: 60000,
+    });
+
+    useEffect(() => {
+        const credentials: CredentialVariables[] | undefined = credentialsQuery.data;
+        if (credentials !== undefined && credentials.length > 0) {
+            setAbstractCredentialInfoList(credentials);
+        } else {
+            setAbstractCredentialInfoList([]);
+        }
+    }, [credentialsQuery.data, credentialsQuery.isSuccess]);
+
+    useEffect(() => {
+        setAbstractCredentialInfoList([]);
+    }, [credentialsQuery.error]);
+
+    const deleteCredentialRequest = useMutation({
```

Comment:
is it possible to move deleteCredential code to separate file?

Reply:
Delete credential codes cannot be moved to a separate file. We need some variables and functions to determine if the deletion was successful, and we also need to update the list. This cannot be done in a separate file. And for the logic code of the delete button in the voucher list, it has no special meaning as a file alone, so I don't think it is necessary to move the delete voucher code to a separate file